### PR TITLE
refactor(linter): run eq_eq_eq fix in some condition

### PR DESCRIPTION
**Pull request by mysteryven**

### CODE REVIEW

File: crates/oxc_linter/src/rules/eslint/eq_eq_eq.rs [hunk 1]
```diff
@@ -91,5 +127,17 @@ fn test() {
         ("value == undefined", None),
     ];
 
-    Tester::new(EqEqEq::NAME, pass, fail).test_and_snapshot();
+    let fix =
+        vec![("null==null", "null===null", None),
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/eslint/eq_eq_eq.rs:131]: we don't have span in operator, is that better to add fixed space like `null === null` ?

**Review by Boshen — COMMENTED** [review]
The `Tester` is getting complicated, we should refactor it a bit to accept fixes. Its API is getting confusing.

**Comment by Boshen** [comment]
> The `Tester` is getting complicated, we should refactor it a bit to accept fixes. Its API is getting confusing.

It's not too late to change all the call sites, but let's do the refactoring in another PR.

**Comment by mysteryven** [comment]
I made some mistakes in this PR, will fix it evening or tomorrow.

File: crates/oxc_linter/src/rules/eslint/eq_eq_eq.rs [hunk 2]
```diff
@@ -34,62 +74,170 @@ declare_oxc_lint!(
...
+        //  - Comparing two literal values
+        //  - Evaluating the value of typeof
+        //  - Comparing against null
+        if matches!(self.compare_type, CompareType::Smart)
+            && (is_type_of_binary_bool || are_literals_and_same_type_bool || is_null)
+        {
+            return;
+        }
 
-        if !is_valid_comparison {
```
Comment (mysteryven) [inline crates/oxc_linter/src/rules/eslint/eq_eq_eq.rs:64]: I need to run fixer in some situation which contained in `is_valid_comparison == true`. so I refactor this rule.

File: crates/oxc_linter/src/rules/eslint/eq_eq_eq.rs [hunk 3]
```diff
@@ -1,21 +1,61 @@
...
+
+impl NullType {
+    pub fn from(raw: &str) -> Self {
+        match raw {
+            "always" => Self::Always,
+            "never" => Self::Never,
+            _ => Self::Ignore,
+        }
+    }
+}
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/eslint/eq_eq_eq.rs:52]: nit: move these code after `impl Rule for EqEqEq` to make this file easier to ready

**Review by Boshen — COMMENTED** [review]
Fantastic work, you may merge yourself after the one tiny clean up.

**Comment by mysteryven** [comment]
> If a collaborator with admin, owner, or write access to the repository submits a review requesting changes, the pull request cannot be merged until the same collaborator submits another review approving the changes in the pull request

I can't merge myself.😢

**Review by DonIsaac — APPROVED** [review]

**Pull request merged**